### ISSUE TITLE
Problem when checking for devextreme-themebuilder npm package version

### ISSUE THREAD

**Issue by echalone** [DevExpress/devextreme-cli#366]
# Bug Report

**Package version:**

    devextreme-cli version: 1.2.15 but should occur with latest source code 1.2.17 too.

**Steps to reproduce:**
When we call yarn install to install all node modules, the latest devextreme-cli version 1.2.15 is installed as well as devextreme-themebuilder version 21.1.3 via devextreme-cli method installThemeBuilder. Both modules are installed correctly (at least after the fact). However, devextreme-cli tries to verify the devextreme-themebuilder version is above version 18.2.5, or at least it says so, in reality it's just catching any error from the installThemeBuilder method but doesn't output what that error was and instead just says the verification for an installed devextreme-themebuilder module version above 18.2.5 failed. This verification fails on just one Computer of ours, not on others, and I can't figure out why. It might have something to do with timing or an error is thrown which I don't see since all it tells me is "The devextreme-themebuilder npm package of v21.1.3 was not installed. Please verify you are using v18.2.5 or higher and examine the installation error log to further troubleshoot the issue." and halts (which is bad for automated builds). There is no installation error log, which makes sense, since the installation actually worked. But, again, if I look at the node_modules folder afterwards, the correct devextreme-themebuilder module of version 21.1.3 IS installed there (in the "root" node_modules folder as well as in the devextreme-cli module-specific node_modules sub-folder). And now to the really fun part: If I run yarn install a second time, it works, as long as the node_modules folder isn't deleted beforehand. That makes me think it might be a timing problem, with maybe some check being performed faster than the file system can create the module files or similar. So sadly, I cannot tell you how to reproduce this issue since it only happens reliably on one Computer of ours (which actually ain't that slow), but my fear is it might not stay that way and may happen on other Computers as well.

The problem seems to be with the installThemeBuilder method in the file packages/devextreme-cli/src/themebuider.js

**Current behavior:**
May tell you a version of the devextreme-themebuilder module above 18.2.5 is not installed.

**Expected behavior:**
Not to fail that check, since the module actually is installed if you look at it afterwards.

**Screenshots:**
When I run yarn install the first time on the affected Computer, it doesn't work and holds with this error message:
![grafik](https://user-images.githubusercontent.com/6392389/123952051-c56cb880-d9a5-11eb-85de-f1f43f84a885.png)

When I cancel execution with CTRL+C and run yarn install a second time it does work:
![grafik](https://user-images.githubusercontent.com/6392389/123952463-5348a380-d9a6-11eb-911e-eeab29b16664.png)

**Comment by babich-a** [comment]
Hi, can you specify a nodejs version?

**Comment by echalone** [comment]
Yes, of course, sorry! Should have thought about that myself ^^
These are our values when entering "npm version" on all the systems, where it works and where it doesn't work (but I also had the same problem when trying with node 16.3.0):
```
{
  npm: '7.19.0',
  node: '14.17.1',
  v8: '8.4.371.23-node.67',
  uv: '1.41.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.17.1',
  modules: '83',
  nghttp2: '1.42.0',
  napi: '8',
  llhttp: '2.1.3',
  openssl: '1.1.1k',
  cldr: '39.0',
  icu: '69.1',
  tz: '2021a',
  unicode: '13.0'
}
```

**Comment by babich-a** [comment]
We published devextreme-cli@1.2.18 that prints the output of `yarn add devextreme-themebuilder`. I hope it will allow you to see the actual error.

**Comment by echalone** [comment]
Thank you! First off, the error doesn't block the execution any longer, the lock.release(); in the catch block seems to do that trick I assume, so thanks for that :)
Second, this is the error I get printed out now (I replaced the actual path to our project with ``<PathToOurProject>`` for security reasons):
```
warning "devextreme-themebuilder > bootstrap@4.6.0" has unmet peer dependency "popper.js@^1.16.1".
[5/5] Building fresh packages...
error <PathToOurProject>\node_modules\devextreme-cli\node_modules\puppeteer: Command failed.
Exit code: 1
Command: node install.js
Arguments:
Directory: <PathToOurProject>\node_modules\devextreme-cli\node_modules\puppeteer
Output:
ERROR: Failed to download Chromium r722234! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.
Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1514:34)
    at TLSSocket.emit (events.js:375:28)
    at TLSSocket._finishInit (_tls_wrap.js:936:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:708:12)
  -- ASYNC --
    at BrowserFetcher.<anonymous> (<PathToOurProject>\node_modules\devextreme-cli\node_modules\puppeteer\lib\helper.js:111:15)
    at Object.<anonymous> (<PathToOurProject>\node_modules\devextreme-cli\node_modules\puppeteer\install.js:66:16)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY'
}
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
The devextreme-themebuilder npm package of v21.1.3 was not installed. Please verify you are using v18.2.5 or higher and examine the installation error log to further troubleshoot the issue.
```
This told me all I needed to know and I immediately knew what's the problem :D Our IT department use their own certificates on top of the websites' certificates. This means we have to tell NodeJS either about our own certificates or to ignore TLS/SSL verification completely. I have previously done this on our build agents, but apparently never did it on my own PC where the problem then appeared. The error means since NodeJS doesn't know about our certificates but requires a TSL connection it cannot download Chromium in the puppeteer module. But this can easily be fixed.
For others who have that same problem in the future: to tell NodeJS about your certificates (on Windows) you have to create a PEM certificate chain (should be easily possible with FireFox) and set the environment variable "NODE_EXTRA_CA_CERTS" to the path of the certificate chain file (preferable machine wide, so you can be sure every process knows about it). You can do this with the following command in PowerShell or PowerShell Core:
``[System.Environment]::SetEnvironmentVariable("NODE_EXTRA_CA_CERTS", "<PathToPEMCertificateChainFile>", "Machine")``
Alternatively you can also tell node to just completely ignore SSL/TSL verification by setting the environment variable "NODE_TLS_REJECT_UNAUTHORIZED" to 0 with the following command in PowerShell or PowerShell Core:
``[System.Environment]::SetEnvironmentVariable("NODE_TLS_REJECT_UNAUTHORIZED", 0, "Machine")``

Thank you for your fast help to the DevExpress/DevExtreme team! I think the ticket can be closed now :)
